tensorflow/core/common_runtime/executor.cc

```diff
@@ -2093,8 +2093,11 @@ void ExecutorState::PropagateOutputs(const TaggedNode& tagged_node,
     // Fast path for nodes types that don't need special handling
     DCHECK_EQ(input_frame, output_frame);
     // Normal path for most nodes
+    {
```

Comment:
note that on line 2094, we check that `input_frame` and `output_frame` point to the same `FrameState` object

your change make the joint update to `output_frame` and `input_frame` non-atomic ... in other words, it introduces an opportunity for a concurrency error